libparsec/crates/protocol/tests/authenticated_cmds/v4/shamir_recovery_setup.rs

```diff
@@ -0,0 +1,236 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+#![allow(clippy::unwrap_used)]
+
+use libparsec_protocol::authenticated_cmds::v4::shamir_recovery_setup::ShamirRecoverySetup;
+use libparsec_tests_lite::prelude::*;
+use libparsec_types::{DateTime, ShamirRevealToken};
+
+use super::authenticated_cmds;
+
+macro_rules! test_de_serialization {
+    ($encoded:literal, $expected:path, $load:path) => {
+        let raw = hex!($encoded);
+        let data = $load(&raw).unwrap();
+        p_assert_eq!(
+            data,
+            $expected,
+            "Expected hex is:\n\"{}\"",
+            encode_and_format_with_70_width($expected.dump().unwrap())
+        );
+
+        // Also test serialization round trip
+        let raw2 = data.dump().unwrap();
+        let data2 = $load(&raw2).unwrap();
+        p_assert_eq!(data2, $expected);
+    };
+}
+
+fn encode_and_format_with_70_width<T: AsRef<[u8]>>(data: T) -> String {
+    hex::encode(data)
+        .as_bytes()
+        .chunks(70)
+        .map(std::str::from_utf8)
+        .collect::<Result<Vec<&str>, _>>()
+        .unwrap()
+        .join(" \\\n")
+}
```

Comment:
This could also be used in other tests

Reply:
Indeed! Most, if not all, tests follow the same pattern. Can you create an issue for that?

Reply:
See #7549